tests/src/JIT/Performance/CodeQuality/BenchmarksGame/k-nucleotide/k-nucleotide.cs

```diff
@@ -0,0 +1,312 @@
+        ulong rollingKey = 0;
+        ulong mask = 0;
+        int cursor;
+        for (cursor = 0; cursor < fragmentLength - 1; cursor++)
+        {
+            rollingKey <<= 2;
+            rollingKey |= tonum[buffer[cursor]];
+            mask = (mask << 2) + 3;
+        }
+        mask = (mask << 2) + 3;
+        int stop = buffer.Length;
+        Wrapper w;
+        byte cursorByte;
+        while (cursor < stop)
+        {
+            if ((cursorByte = buffer[cursor++]) < (byte)'a')
+                cursorByte = buffer[cursor++];
+            rollingKey = ((rollingKey << 2) & mask) | tonum[cursorByte];
+            if (dictionary.TryGetValue(rollingKey, out w))
+                w.V++;
+            else
+                dictionary.Add(rollingKey, new Wrapper(1));
+        }
+        return dictionary;
+    }
+
+    private static byte[] GetBytesForThirdSequence(FileStream source)
+    {
+        const int buffersize = 2500120;
+        byte[] threebuffer = null;
+        var buffer = new byte[buffersize];
+        int amountRead, threebuflen, indexOfFirstByteInThreeSequence, indexOfGreaterThan, threepos, tocopy;
+        amountRead = threebuflen = indexOfFirstByteInThreeSequence = indexOfGreaterThan = threepos = tocopy = 0;
+        bool threeFound = false;
+        //var source = new FileStream(inputFile, FileMode.Open);
+        source.Seek(0, SeekOrigin.Begin);
+        while (!threeFound && (amountRead = source.Read(buffer, 0, buffersize)) > 0)
+        {
+            indexOfGreaterThan = Array.LastIndexOf(buffer, (byte)'>');
+            threeFound = (indexOfGreaterThan > -1 &&
+                buffer[indexOfGreaterThan + 1] == (byte)'T' &&
+                buffer[indexOfGreaterThan + 2] == (byte)'H');
+            if (threeFound)
+            {
+                threepos += indexOfGreaterThan;
+                threebuflen = threepos - 48;
+                threebuffer = new byte[threebuflen];
+                indexOfFirstByteInThreeSequence = Array.IndexOf<byte>(buffer, 10, indexOfGreaterThan) + 1;
+                tocopy = amountRead - indexOfFirstByteInThreeSequence;
+                if (amountRead < buffersize)
+                    tocopy -= 1;
+                Buffer.BlockCopy(buffer, indexOfFirstByteInThreeSequence, threebuffer, 0, tocopy);
+                buffer = null;
+            }
+            else
+                threepos += amountRead;
+        }
+        int toread = threebuflen - tocopy;
+        source.Read(threebuffer, tocopy, toread);
+        return threebuffer;
+    }
+
+    private static byte[] tonum = new byte[256];
+    private static char[] tochar = new char[4];
+    private static void PrepareLookups()
+    {
+        tonum['a'] = 0;
+        tonum['c'] = 1;
+        tonum['g'] = 2;
+        tonum['t'] = 3;
+        tochar[0] = 'A';
+        tochar[1] = 'C';
+        tochar[2] = 'G';
+        tochar[3] = 'T';
+    }
+
+    [Benchmark(InnerIterationCount=Iterations)]
+    public static void Bench()
```

Comment:
Can you confirm that these methods show up with nice names in Benchview, so that it's clear what they are?